include/deal.II/base/symmetric_tensor.h

```diff
@@ -1667,13 +1694,42 @@ SymmetricTensor<rank,dim,Number>::operator [] (const TableIndices<rank> &indices
 
 
 
+
+namespace internal
+{
+  namespace SymmetricTensor
+  {
+    template <int dim, typename Number>
+    unsigned int
+    entry_to_indices (const dealii::SymmetricTensor<2,dim,Number> &,
```

Comment:
But I need overloads for the two different ranks (that also return different types), and I can't do that as you suggest because there is no such thing as a partial specialization of function templates. It really doesn't work without the unused argument (unless I want to enclose the function into a class of its own, because there are partial specializations of class templates; but then I still have to write a separate dispatch class to deal with the different return types).